Source/santad/SNTPolicyProcessor.m

```diff
@@ -0,0 +1,166 @@
+/// Copyright 2015 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#import "SNTPolicyProcessor.h"
+
+#include "SNTLogging.h"
+
+#import "SNTCachedDecision.h"
+#import "SNTConfigurator.h"
+#import "SNTFileInfo.h"
+#import "SNTRule.h"
+#import "SNTRuleTable.h"
+
+@interface SNTPolicyProcessor()
+@property SNTRuleTable *ruleTable;
+@end
+
+@implementation SNTPolicyProcessor
+
+- (instancetype)initWithRuleTable:(SNTRuleTable *)ruleTable {
+  self = [super init];
+  if (self) {
+    _ruleTable = ruleTable;
+  }
+  return self;
+}
+
+- (SNTCachedDecision *)decisionForFileInfo:(SNTFileInfo *)fileInfo
+                                fileSHA256:(NSString *)fileSHA256
+                        signingCertificate:(MOLCertificate *)signingCertificate {
+  SNTCachedDecision *cd = [[SNTCachedDecision alloc] init];
+  if (fileInfo) {
```

Comment:
If there's no fileInfo, this should fail somehow; without the SHA-256 we shouldn't continue.